test/studies/IMSuite/vertexColoring.chpl

```diff
@@ -0,0 +1,323 @@
+    http://www.opensource.org/licenses/eclipse-1.0.php
+
+   (C) Copyright IMSuite 2013-present.
+ */
+
+ /*
+    vertexColoring aims to color the vertices of a tree with
+    three colors.
+
+    @author Suyash Gupta
+    @author V Krishna Nandivada
+
+    Ported By: Kushal Singh
+ */
+
+module vertexColoring {
+
+    use Random;
+    use VisualDebug;
+
+    config var inputFile = "inputvertexColoring_16_max.txt";
+    config var outputFile = "outputvertexColoring.txt";
+    config var verify = false;
+
+    var nodes: int;
+    var colorLabel: int;
+    var D : domain(1);
+    var adj_graph :[D][D] int;
+    var parent:[D] int;
+    var root: int;
+    var nlabel:[D] int;
+    var colormat:[D] int;
+
+    /* Parameters to enable execution with load */
+    config var loadValue=0;
+    var nval:[D] int;
+
+    class Node {
+      var parent: int;                  // specifies the parent of a node
+      var color: int;                   // specifies the color of a node
+      var receivedColor: int;           // specifies the color received by a node
+      var children: [1..0] int;         // list of neighbor ids
+    }
+
+    /** Abstract node representation */
+    var nodeSet:[D] Node;
+    var again:[D] bool;
+
+    /*
+        Acts as the starting point for the program execution.
+        ```main``` performs the task of accepting the input from the user
+        specified file, electing the leader and transmitting information,
+        printing the output and validating the result.
+    */
+    proc main() {
+
+        var inFile = open(inputFile, iomode.r);
+        var reader = inFile.reader();
+
+        nodes = reader.read(int);
+        D = {0..(nodes-1)};
+        root = reader.read(int);
+        var line:string;
+
+        for i in D {
+            reader.read(line);
+            for j in 0..(line.length-1) {
+                adj_graph[i][j]=(line[j+1]:int);
+            }
+        }
+
+        for i in D {
+            parent[i]=reader.read(int);
+        }
+
+        reader.close();
+        inFile.close();
+
+        initialize();
+        run();
+        six2three();
+        printOutput();
+
+        if(verify) then
+            outputVerifier();
+
+        if(loadValue != 0) {
+			var sumval: real = 0;
+            for i in D{
+                    sumval = sumval + nval[i];
+            }
+			if(sumval > 0) then
+                writeln();
+		}
+    }
+
+    /* Initializes all the fields of the abstract node. */
+    proc initialize() {
+       forall i in D {
+            nlabel[i] = i;
+            nodeSet[i] =  new Node();
+            nodeSet[i].parent = parent[i];
+            forall j in D {
+                if(adj_graph[i][j] == 1 && nodeSet[i].parent != j) then
+                    nodeSet[i].children.push_back(j);
+            }
+            nodeSet[i].color = nlabel[i];
+        }
+        colorLabel = log2(nodes);
+        if(1<<colorLabel < nodes) then
+            colorLabel+=1;
+    }
```

Comment:
Instead of indexing each individual array, it could be cleaner to use zippered iteration, like this:
```chpl
forall (node, label, i) in zip(nodeSet, nlabel, D) { 
  // node == nodeSet[i]
}
```
You can read more about parallel zippered iterators in examples/primers/parIters.chpl